### PR TITLE
Profile tab revamp

### LINKED ISSUE #20: Add user profile widget
- [x] create a separate page for it with an AppBar
- [x] render posts
- [x] render comments
- [x] render about
    - [x] bio
	- [x] moderates
	- [x] subscribed
- [x] fix bug where the banner can overlap the navigation pill

Profile_mockup_

not to be confused with #19

### PR DESCRIPTION
**Pull request by shilangyu**
closes #20 

I hope my sins for the previous implementation can be forgiven

recordedVideo-compressed

### CODE REVIEW

File: lib/widgets/user_profile.dart [hunk 1]
```diff
@@ -5,283 +5,382 @@ import 'package:intl/intl.dart';
...
-                ),
-              )
-            ],
+    return DefaultTabController(
+      length: 3,
+      child: NestedScrollView(
+        headerSliverBuilder: (_, __) => [
+          SliverAppBar(
+            pinned: true,
+            expandedHeight: 255,
```
Comment (krawieck) [inline lib/widgets/user_profile.dart:48]: increase expanded height cuz cake day date is too close to tab titles
image
the space should be about the same as the one between user badges and "joined ..." text

File: lib/widgets/user_profile.dart [hunk 2]
```diff
@@ -5,283 +5,382 @@ import 'package:intl/intl.dart';
...
+              child:
+                  MarkdownText(userDetails.user.bio, instanceUrl: instanceUrl)),
+          divider,
         ],
-      ),
+        if (userDetails.moderates.isNotEmpty) ...[
+          Padding(
+            padding: wallPadding,
+            child: Text('Moderates', style: theme.textTheme.subtitle2),
+          ),
```
Comment (krawieck) [inline lib/widgets/user_profile.dart:351]: section titles should be based on the ones in [instance page](https://github.com/krawieck/lemmur/blob/1eb3c9e937ec042f832b55b9cb9f886ca1b1c4f7/lib/pages/instance.dart#L347-L354), centered and  based on headline6 with fontSize set to

File: lib/widgets/user_profile.dart [hunk 3]
```diff
@@ -5,283 +5,382 @@ import 'package:intl/intl.dart';
...
+              title: Text('!${comm.communityName}'),
+              onTap: () =>
+                  goToCommunity.byId(context, instanceUrl, comm.communityId),
+            ),
+          divider
+        ],
+        Padding(
+          padding: wallPadding,
+          child: Text('Subscribed', style: theme.textTheme.subtitle2),
+        ),
```
Comment (krawieck) [inline lib/widgets/user_profile.dart:364]: section titles should be based on the ones in [instance page](https://github.com/krawieck/lemmur/blob/1eb3c9e937ec042f832b55b9cb9f886ca1b1c4f7/lib/pages/instance.dart#L347-L354), centered and  based on headline6 with fontSize set to

File: lib/widgets/user_profile.dart [hunk 4]
```diff
@@ -5,283 +5,382 @@ import 'package:intl/intl.dart';
...
+          Padding(
+            padding: const EdgeInsets.only(top: 8),
+            child: Center(
+              child: Text(
+                'this user does not subscribe to any community',
+                style: TextStyle(fontStyle: FontStyle.italic),
+              ),
+            ),
+          )
+      ],
```
Comment (krawieck) [inline lib/widgets/user_profile.dart:383]: this doesn't seem to work. on every user i check it says that _this user does not subscribe to any community_

File: lib/widgets/user_profile.dart [hunk 5]
```diff
@@ -5,283 +5,382 @@ import 'package:intl/intl.dart';
...
+            padding: wallPadding,
+            child: Text('Moderates', style: theme.textTheme.subtitle2),
+          ),
+          for (final comm in userDetails.moderates)
+            ListTile(
+              dense: true,
+              title: Text('!${comm.communityName}'),
+              onTap: () =>
+                  goToCommunity.byId(context, instanceUrl, comm.communityId),
+            ),
```
Comment (krawieck) [inline lib/widgets/user_profile.dart:358]: list of moderated communities should have community icons like [list of trending communities in instance page](https://github.com/krawieck/lemmur/blob/1eb3c9e937ec042f832b55b9cb9f886ca1b1c4f7/lib/pages/instance.dart#L357-L376)

File: lib/widgets/user_profile.dart [hunk 6]
```diff
@@ -5,283 +5,382 @@ import 'package:intl/intl.dart';
...
+          child: Text('Subscribed', style: theme.textTheme.subtitle2),
+        ),
+        if (userDetails.follows.isEmpty)
+          for (final comm in userDetails.follows)
+            ListTile(
+              dense: true,
+              title: Text('!${comm.communityName}'),
+              onTap: () =>
+                  goToCommunity.byId(context, instanceUrl, comm.communityId),
+            )
```
Comment (krawieck) [inline lib/widgets/user_profile.dart:372]: list of followed communities should have community icons like [list of trending communities in instance page](https://github.com/krawieck/lemmur/blob/1eb3c9e937ec042f832b55b9cb9f886ca1b1c4f7/lib/pages/instance.dart#L357-L376)

File: lib/widgets/user_profile.dart [hunk 7]
```diff
@@ -5,283 +5,382 @@ import 'package:intl/intl.dart';
...
+
+    return ListView(
+      padding: EdgeInsets.symmetric(vertical: 20),
+      children: [
+        if (isOwnedAccount)
+          Center(
+            child: FlatButton.icon(
+              icon: Icon(Icons.edit),
+              label: Text('edit profile'),
+              onPressed: () {}, // TODO: go to account editing
```
Comment (krawieck) [inline lib/widgets/user_profile.dart:337]: this button should be a ListTile with onTap and centered text like in [instance page](https://github.com/krawieck/lemmur/blob/1eb3c9e937ec042f832b55b9cb9f886ca1b1c4f7/lib/pages/instance.dart#L388-L391)

**Review by krawieck — CHANGES_REQUESTED** [review]
the theme of these comments is that you should base UI on instance page, and not community page. community page was the one i made earlier so i wasn't entirely sure about design. i should probably update it...

File: lib/widgets/user_profile.dart (hunk 4 above)
Comment (shilangyu) [reply lib/widgets/user_profile.dart:383]: ah yes, I changed it for debug and forgot to revert

**Review by krawieck — APPROVED** [review]

**Pull request merged**